CMA Build Warnings

Recently noticed these in the Travis CMA build.  The ```strerror_r``` warnings are likely because ```_GNU_SOURCE``` changes the prorotype for this function (https://linux.die.net/man/3/strerror_r).  We may want to just add a ```shmem_internal_strerror_r``` routine to wrap this API shift.  I wonder if we want to just set ```_GNU_SOURCE``` globally rather than potentially having a different prototype for this function in different compilation units..
```
make[1]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
  GEN      atomic_c.c
  GEN      collectives_c.c
  GEN      data_c.c
  GEN      synchronization_c.c
  GEN      collectives_f.c
  GEN      data_f.c
  GEN      build_info.h
make  all-am
make[2]: Entering directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
  CC       shmem_free_list.lo
  CC       malloc.lo
  CC       init.lo
  CC       collectives.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_synchronization.h:20,
                 from ../../src/shmem_collectives.h:19,
                 from ../../src/init.c:30:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_synchronization.h:20,
                 from ../../src/shmem_collectives.h:19,
                 from ../../src/collectives.c:22:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       init_c.lo
  CC       query_c.lo
  CC       accessibility_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_accessibility.h:19,
                 from ../../src/accessibility_c.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       symmetric_heap_c.lo
  CC       remote_pointer_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/symmetric_heap_c.c:34:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_remote_pointer.h:19,
                 from ../../src/remote_pointer_c.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       lock_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_lock.h:19,
                 from ../../src/lock_c.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       cache_management_c.lo
  CC       util.lo
  CC       shmem_env.lo
  CC       contexts_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_synchronization.h:20,
                 from ../../src/contexts_c.c:22:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       init_f.lo
  CC       query_f.lo
  CC       accessibility_f.lo
  CC       symmetric_heap_f.lo
  CC       remote_pointer_f.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_accessibility.h:19,
                 from ../../src/accessibility_f.c:19:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       atomic_f.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_synchronization.h:20,
                 from ../../src/shmem_collectives.h:19,
                 from ../../src/symmetric_heap_f.c:23:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_remote_pointer.h:19,
                 from ../../src/remote_pointer_f.c:21:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/atomic_f.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       synchronization_f.lo
  CC       lock_f.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/synchronization_f.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       cache_management_f.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_lock.h:19,
                 from ../../src/lock_f.c:24:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       transport_none.lo
  CC       transport_cma.lo
  CC       runtime-pmi.lo
  CC       atomic_c.lo
  CC       collectives_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from atomic_c.c:28:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/transport_cma.c:30:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
In file included from ../../src/shmem_comm.h:49:0,
                 from collectives_c.c:30:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       data_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from data_c.c:30:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       synchronization_c.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from ../../src/shmem_synchronization.h:20,
                 from synchronization_c.c:29:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       collectives_f.lo
In file included from ../../src/shmem_comm.h:49:0,
                 from collectives_f.c:31:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CC       data_f.lo
  SED      oshcc
  SED      oshrun
  SED      oshCC
  SED      oshc++
  SED      oshfort
In file included from ../../src/shmem_comm.h:49:0,
                 from data_f.c:30:
../../src/transport_cma.h: In function ‘shmem_transport_cma_put’:
../../src/transport_cma.h:108:9: warning: implicit declaration of function ‘process_vm_writev’ [-Wimplicit-function-declaration]
         bytes = process_vm_writev( target_pid,
         ^
../../src/transport_cma.h:115:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
../../src/transport_cma.h: In function ‘shmem_transport_cma_get’:
../../src/transport_cma.h:144:9: warning: implicit declaration of function ‘process_vm_readv’ [-Wimplicit-function-declaration]
         bytes = process_vm_readv(target_pid,
         ^
../../src/transport_cma.h:150:28: warning: initialization makes pointer from integer without a cast [enabled by default]
             char *errstr = strerror_r(errno, errmsg, 128);
                            ^
  CCLD     libsma.la
make[2]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
make[1]: Leaving directory `/home/travis/build/Sandia-OpenSHMEM/SOS/cma-build/src'
```

--- a/configure.ac
+++ b/configure.ac
@@ -341,6 +341,7 @@ elif test -n "$with_xpmem" -a "$with_xpmem" != "no" ; then
 elif test -n "$with_cma" -a "$with_cma" != "no" ; then
     transport_xpmem="no"
     AC_DEFINE([USE_CMA], [1], [Define if Cross Memory Attach transport is active])
+    AC_DEFINE([_GNU_SOURCE], [1], [CMA transport header requires global definition of _GNU_SOURCE])
 # If neither, disable XPMEM and CMA:
 else
     transport_xpmem="no"

--- a/src/transport_cma.h
+++ b/src/transport_cma.h
@@ -17,7 +17,7 @@
 #define SHMEM_TRANSPORT_CMA_H
 
 #ifndef _GNU_SOURCE
-#define _GNU_SOURCE
+#error CMA transport requires definition of _GNU_SOURCE
 #endif
 
 #include <sys/uio.h>
@@ -115,7 +115,7 @@ shmem_transport_cma_put(void *target, const void *source, size_t len,
 #else
             char *errstr = errmsg;
             int err = strerror_r(errno, errmsg, 128);
-            if (err) RAISER_ERROR_MSG("Error in call to strerr_r (%d)\n", err);
+            if (err) RAISE_ERROR_MSG("Error in call to strerr_r (%d)\n", err);
 #endif
             RAISE_ERROR_MSG("process_vm_writev() failed (%s)\n", errstr);
         }
@@ -149,8 +149,8 @@ shmem_transport_cma_get(void *target, const void *source, size_t len, int pe,
             char *errstr = strerror_r(errno, errmsg, 128);
 #else
             char *errstr = errmsg;
-            err = strerror_r(errno, errmsg, 128);
-            if (err) RAISER_ERROR_MSG("Error in call to strerr_r (%d)\n", err);
+            int err = strerror_r(errno, errmsg, 128);
+            if (err) RAISE_ERROR_MSG("Error in call to strerr_r (%d)\n", err);
 #endif
             RAISE_ERROR_MSG("process_vm_readv() failed (%s)\n", errstr);
         }

--- a/src/transport_cma.h
+++ b/src/transport_cma.h
@@ -20,9 +20,8 @@
 #define _GNU_SOURCE
 #endif
 
-#ifdef HAVE_LIBC_CMA
 #include <sys/uio.h>
-#else
+#ifndef HAVE_LIBC_CMA
 #include <sys/syscall.h>
 #endif
 

--- a/src/transport_ofi.c
+++ b/src/transport_ofi.c
@@ -10,6 +10,14 @@
  */
 
 #include "config.h"
+
+#ifdef HAVE_SYS_GETTID
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+#include <sys/syscall.h>
+#endif
+
 #include <errno.h>
 #include <stdlib.h>
 #include <sys/time.h>
@@ -21,13 +29,6 @@
 #include <stdint.h>
 #include <inttypes.h>
 #include <netdb.h>
-
-#ifdef HAVE_SYS_GETTID
-#ifndef _GNU_SOURCE
-#define _GNU_SOURCE
-#endif
-#include <sys/syscall.h>
-#endif
 
 #if HAVE_FNMATCH_H
 #include <fnmatch.h>

